frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/PipelineTaskNode.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { PopoverProps } from '@patternfly/react-core';
+import {
+  DEFAULT_LAYER,
+  DEFAULT_WHEN_OFFSET,
+  Layer,
+  Node,
+  ScaleDetailsLevel,
+  TaskNode,
+  TOP_LAYER,
+  useDetailsLevel,
+  useHover,
+  WhenDecorator,
+  WithContextMenuProps,
+  WithSelectionProps,
+} from '@patternfly/react-topology';
+import { observer } from 'mobx-react';
+import { Link } from 'react-router-dom';
+import { resourcePathFromModel } from '@console/internal/components/utils';
+import { PipelineRunModel } from '../../../models';
+import { ComputedStatus } from '../../../types';
+import {
+  createStepStatus,
+  StepStatus,
+} from '../detail-page-tabs/pipeline-details/pipeline-step-utils';
+import { PipelineVisualizationStepList } from '../detail-page-tabs/pipeline-details/PipelineVisualizationStepList';
+
+type PipelineTaskNodeProps = {
+  element: Node;
```

Comment:
Missing the type here, but it looks like there is no type for the data that has a `task`, `taskIconClass`, `taskIcon`, etc. So we can not add it now, but we should create and request such types in the future.

Reply:
Yes we can not add it now, the base ElementModel's data is mapped to any. so we don't have any type to add here at the moment.